src/core/credrank/testUtils.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import deepFreeze from "deep-freeze";
+import {
+  NodeAddress as NA,
+  EdgeAddress as EA,
+  type Node as GraphNode,
+  type Edge as GraphEdge,
+  Graph,
+} from "../graph";
+import {type Weights as WeightsT} from "../weights";
+import {type WeightedGraph} from "../weightedGraph";
+import {
+  type NodeWeightEvaluator,
+  type EdgeWeightEvaluator,
+  nodeWeightEvaluator,
+  edgeWeightEvaluator,
+} from "../algorithm/weightEvaluator";
+import {
+  type Participant,
+  type Parameters as MarkovProcessGraphParameters,
+  type Arguments as MarkovProcessGraphArguments,
+  MarkovProcessGraph,
+} from "./markovProcessGraph";
+import * as uuid from "../../util/uuid"; // for spy purposes
+import {type IntervalSequence, intervalSequence} from "../interval";
+import {markovProcessGraphPagerank} from "./compute";
+import {CredGraph} from "./credGraph";
+
+/**
+ * This module contains test helpers for working with CredRank data,
+ * specifically MarkovProcessGraphs and CredGraphs
+ */
+
+const na = (name) => NA.fromParts([name]);
+const ea = (name) => EA.fromParts([name]);
+
+export const participantNode1: GraphNode = {
+  description: "participant1",
+  address: na("participant1"),
+  timestampMs: null,
+};
+export const participantNode2: GraphNode = {
+  description: "participant2",
+  address: na("participant2"),
+  timestampMs: null,
+};
+deepFreeze([participantNode1, participantNode2]);
+
+export const participant1: Participant = {
+  description: participantNode1.description,
+  address: participantNode1.address,
+  id: uuid.fromString("YVZhbGlkVXVpZEF0TGFzdA"),
+};
+export const participant2: Participant = {
+  description: participantNode2.description,
+  address: participantNode2.address,
+  id: uuid.fromString("YVZhbGlkVXVpZE20TGFzdA"),
+};
+
+export const participants: $ReadOnlyArray<Participant> = deepFreeze([
+  participant1,
+  participant2,
+]);
+
+const intervals: IntervalSequence = deepFreeze(
+  intervalSequence([
+    {startTimeMs: 0, endTimeMs: 2},
+    {startTimeMs: 2, endTimeMs: 4},
+  ])
+);
+
+const c0 = {description: "c0", address: na("c0"), timestampMs: 0};
+const c1 = {description: "c1", address: na("c1"), timestampMs: 2};
+export const contributions: $ReadOnlyArray<GraphNode> = deepFreeze([c0, c1]);
+
+// Connects contribution 0 to participant 1
+export const e0: GraphEdge = {
+  address: ea("e0"),
+  src: c0.address,
+  dst: participantNode1.address,
+  timestampMs: 1,
+};
+
+// Connects contribution 1 to participant 1
+export const e1: GraphEdge = {
+  address: ea("e1"),
+  src: c1.address,
+  dst: participantNode1.address,
+  timestampMs: 3,
+};
+
+// Connects c0 to c1.
```

Comment:
nit: If you're writing the comments anyways, doc comments instead of // comments enable sweet IDE visibility that improves developer speed, even (and maybe especially) on silly test utils like this.

Reply:
Ack, will do in future.